### PR TITLE
Split functions configuration from FeaturesConfig

### PR DESCRIPTION
**Pull request by tdcmeehan**
## Description
FeaturesConfig has dozens of properties, many unrelated to functions.  In order to simplify the dependencies to `BuiltInTypeAndFunctionNamespaceManager`, I've split up the function related config properties to a new config object dedicated to function properties.

## Motivation and Context
Refactoring

## Impact
No production impact

## Test Plan
Unit test and product test coverage should be sufficient for this change.

## Contributor checklist

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by elharo — APPROVED** [review]

**Review by ZacBlanco — DISMISSED** [review]

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java [hunk 1]
```diff
@@ -120,12 +116,7 @@ public class FeaturesConfig
     private boolean pushTableWriteThroughUnion = true;
     private boolean exchangeCompressionEnabled;
     private boolean exchangeChecksumEnabled;
-    private boolean legacyArrayAgg;
-    private boolean reduceAggForComplexTypesEnabled = true;
-    private boolean legacyLogFunction;
-    private boolean useAlternativeFunctionSignatures;
     private boolean legacyTimestamp = true;
```
Comment (rschlussel) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java:119]: what about legacyTimestamp, legacyRowFieldOrdinalAccess and legacyCharToVarcharCoercion?

Comment (rschlussel) [reply presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java:119]: also fieldNameInJsonCastEnabled and legacyJsonCast

Comment (tdcmeehan) [reply presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java]: I've added all the session properties that delegate to system properties in `FeaturesConfig` in `Session#getSqlFunctionProperties`, which includes all of the above, and also `legacyCharToVarcharCoercion`.

Comment (tdcmeehan) [reply presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java]: Also `legacyJsonCast`.  All have been added.

**Review by rschlussel — APPROVED** [review]

**Review by hantangwangd — APPROVED** [review]

**Comment by tdcmeehan** [comment]
@rschlussel @ZacBlanco @hantangwangd thanks for your approvals.  I had to resolve a rebase conflict, do you mind approving again?

**Review by rschlussel — APPROVED** [review]

**Review by hantangwangd — APPROVED** [review]

**Pull request merged**